packages/lesswrong/components/ultraFeed/UltraFeedPostItem.tsx

```diff
@@ -0,0 +1,240 @@
+    backgroundColor: theme.palette.panelBackground.default,
+    borderRadius: 4,
+  },
+  header: {
+    display: 'flex',
+    flexDirection: 'column',
+    justifyContent: 'space-between',
+    marginBottom: 16,
+  },
+  titleRow: {
+    display: 'flex',
+    justifyContent: 'space-between',
+    alignItems: 'flex-start',
+    width: '100%',
+    marginBottom: 4,
+  },
+  titleContainer: {
+    flexGrow: 1,
+    paddingRight: 8,
+    marginBottom: 4,
+  },
+  title: {
+    fontFamily: theme.palette.fonts.sansSerifStack,
+    fontSize: '1.6rem',
+    fontWeight: 600,
+    opacity: 0.8,
+    lineHeight: 1.15,
+    textWrap: 'balance',
+    width: '100%',
+    '&:hover': {
+      opacity: 0.9,
+    },
+  },
+  headerRightSection: {
+    display: "flex",
+    flexGrow: 0,
+  },
+  tripleDotMenu: {
+    padding: 5,
+    marginLeft: 4,
+    marginRight: -10,
+    "& svg": {
+      fontSize: "1.4rem",
+      cursor: "pointer",
+      color: theme.palette.text.dim,
+    }
+  },
+  metaRoot: {
+    position: "relative",
+    display: "flex",
+    flexWrap: "wrap",
+    alignItems: "center",
+    rowGap: "6px",
+    color: theme.palette.text.dim,
+    fontFamily: theme.palette.fonts.sansSerifStack,
+    fontSize: "1.3rem",
+    "& > *": {
+      marginRight: 5,
+    },
+    "& a:hover, & a:active": {
+      textDecoration: "none",
+      color: `${theme.palette.linkHover.dim} !important`,
+    },
+  },
+  metaLeftSection: {
+    display: "flex",
+    alignItems: "center",
+    flex: "1 1 auto",
+    flexWrap: "wrap",
+  },
+  metaRightSection: {
+    display: "flex",
+    alignItems: "center",
+    marginLeft: "auto",
+    marginRight: 0,
+  },
+  metaKarma: {
+    display: "inline-block",
+    textAlign: "center",
+    flexGrow: 0,
+    flexShrink: 0,
+    paddingRight: 8,
+    marginRight: 4,
+  },
+  metaUsername: {
+    marginRight: 12,
+    '& a, & a:hover': {
+      color: theme.palette.link.unmarked,
+    },
+    color: theme.palette.text.dim,
+    fontFamily: theme.palette.fonts.sansSerifStack,
+    whiteSpace: 'nowrap',
+    display: 'flex',
+    alignItems: 'center',
+  },
+  metaCoauthors: {
+    marginLeft: 4,
+    marginRight: 0,
+    color: theme.palette.text.dim,
+    whiteSpace: 'nowrap',
+  },
+  metaDateContainer: {
+    marginRight: 8,
+  },
+  footer: {
+    // paddingBottom: 24,
+  },
+}));
+
+const UltraFeedPostItem = ({
+  post,
+  postMetaInfo,
+}: {
+  post: PostsListWithVotes,
+  postMetaInfo: FeedPostMetaInfo,
+}) => {
+  const classes = useStyles(styles);
+  const { PostActionsButton, FeedContentBody, UltraFeedItemFooter, FormatDate, UsersName, UserNameDeleted } = Components;
+  const { settings } = useUltraFeedSettings();
+
+  const { observe, trackExpansion } = useUltraFeedObserver();
+  const elementRef = useRef<HTMLDivElement | null>(null);
+
+  const { isAnon, authors } = usePostsUserAndCoauthors(post);
+  const showKarma = !post.rejected;
+
+  const renderAuthors = () => {
```

Comment:
don't love inline component functions; would prefer just having a tiny component outside of this that uses the `usePostsUserAndCoauthors` hook and returns the appropriate thing.